ui/client/components/graph/NodeDetailsModal.js

```diff
@@ -161,25 +161,54 @@ class NodeDetailsModal extends React.Component {
     return (<BareGraph processCounts={subprocessCounts} processToDisplay={this.state.subprocessContent}/>)
   }
 
-  renderDocumentationIcon() {
-    const docsUrl = this.props.nodeSetting.docsUrl
-    return docsUrl ?
-      <a className="docsIcon" target="_blank" href={docsUrl} title="Documentation">
-        <SvgDiv svgFile={'documentation.svg'}/>
-      </a> : null
-  }
-
   variableLanguage = (node) => {
     return _.get(node, 'value.language')
   }
 
+  renderServiceName = (name, docsUrl) => {
+    if(name) {
+      return docsUrl ?
+        <a className="docsLink" target="_blank" href={docsUrl} title="Documentation">
+          <div className="modal-subtype-header-docs">
+            <span>{name}</span>
+            <SvgDiv className="docsIcon" svgFile={'documentation.svg'}/>
+          </div>
+        </a> :
+        <div className="modal-subtype-header">
+          <span>{name}</span>
+        </div>
+    }
+    else {
+         return docsUrl ?
+           <a className="docsLink" target="_blank" href={docsUrl} title="Documentation">
+             <div className="modal-docs-link">
+               <SvgDiv className="docsIcon" svgFile={'documentation.svg'}/>
+             </div>
+           </a> :
+           null
+    }
+  }
+
+  findOneOfProperties = (obj, props) => {
+    const property = _.find(props, (property) => {
+      return _.has(obj, property)
+    })
+    return property ? _.get(obj, property) : null
+  }
+
   render() {
-    const isOpen = !_.isEmpty(this.props.nodeToDisplay) && this.props.showNodeDetailsModal
+    const {nodeErrors, nodeToDisplay, nodeSetting, readOnly, showNodeDetailsModal, testResults} = this.props
+
+    const isOpen = !_.isEmpty(nodeToDisplay) && showNodeDetailsModal
     const titleStyles = EspModalStyles.headerStyles(this.nodeAttributes().styles.fill, this.nodeAttributes().styles.color)
-    const testResults = (id) => TestResultUtils.resultsForNode(this.props.testResults, id)
-    const variableLanguage = this.variableLanguage(this.props.nodeToDisplay)
+    const testResultsFun = (id) => TestResultUtils.resultsForNode(testResults, id)
+    const variableLanguage = this.variableLanguage(nodeToDisplay)
     const modelHeader = (_.isEmpty(variableLanguage) ? "" : `${variableLanguage} `) + this.nodeAttributes().name
 
+    const properties = ["service.id", "ref.typ", "nodeType", "ref.id"]
+    const serviceID = this.findOneOfProperties(nodeToDisplay, properties)
```

Comment:
Please consult @arkadius and @mproch if this is a correct name. I've not encoutered it before, Arek came up with a "node subtype".